Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -478,3 +493,105 @@ void MainWindow::SetStateSlot(int slot)
   Settings::Instance().SetStateSlot(slot);
   m_state_slot = slot;
 }
+
+void MainWindow::PerformOnlineUpdate(const std::string& region)
```

Comment:
How should this be extracted? Is it supposed to be a member function of MainWindow?